### ISSUE TITLE
Unable to find callback for callAPIMethod method in iOS

### ISSUE THREAD

**Issue by rajaarun** [rdio/api#160]
Previously i was using callApiMethod to get user detail in my application.

 `[[AppDelegate rdioInstance] callAPIMethod:@"findUser"
                                   withParameters:params
                                         delegate:[RDAPIRequestDelegate delegateToTarget:self
                                                                            loadedAction:@selector(rdioRequest:didLoadData:)
                                                                            failedAction:@selector(rdioRequest:didFailWithError:)]];`

Since Rdio has suggested to use **OAuth 2.0**  i had followed the steps mentioned in the link http://www.rdio.com/developers/docs/guides/oauth2-migration/ for ios SDK 

Added rdio-ios-3.0.3 SDK, 
As i am not using pods, i manually downloaded **AFOAuth2Manager** and **AFNetworking** in my project
replaced the `initWithConsumerKey:@"" andSecret:@"" delegate:nil` to `initWithClientId:@"" andSecret:@"" delegate:nil` with new client id and secret key,
and also replaced the callAPIMethod function  
`[[AppDelegate rdioInstance] callAPIMethod:@"findUser"
                                   withParameters:params
                                         delegate:[RDAPIRequestDelegate delegateToTarget:self
                                                                            loadedAction:@selector(rdioRequest:didLoadData:)
                                                                            failedAction:@selector(rdioRequest:didFailWithError:)]];` to

`[[AppDelegate rdioInstance] callAPIMethod:@"findUser" withParameters:params success:^(NSDictionary *result) { } failure:^(NSError *error) { }];`

But i am not able to get the response from any block, neither the success block nor the failure block.

Also had similar issue with  callAPIMethod:@"getHeavyRotation" method too. Also getting this error in console **Unable to fetch credential with identifier "services.rdio.com" (Error -25300)**.

**Comment by dasevilla** [comment]
Could this be an issue with threading? From the documentation

> The Rdio SDK is not currently thread-safe. The current implementation expects to be called from the main thread at all times.

http://www.rdio.com/developers/docs/libraries/ios/ref-an-important-note-on-threading

**Comment by gabek** [comment]
@rajaarun Could you provide a sample project that shows the issue?

**Comment by anupamwcities** [comment]
Yes, i can provide you a sample project. Can you share your email id.

**Comment by rajaarun** [comment]
Yes, [here](https://github.com/rajaarun/RdioDemo)  is the link for sample project.

**Comment by gabek** [comment]
Looks like that Error -25300 might be specific to iOS 9.  There might be some keychain security changes in 9.  Have you tried running your sample project until iOS 8 as well?

**Comment by dasevilla** [comment]
@rajaarun, can you try keeping the `RDAPIRequest` reference returned by `callAPIMethod` until after the success or failure blocks have run?

**Comment by dasevilla** [comment]
As of [v3.1.0](http://blog.rdio.com/developers/2015/10/ios-sdk-v310.html), you should receive a callback
